src/main/java/com/three_tech_solutions/slot_app/services/implementations/StudentServiceImpl.java

```diff
@@ -124,13 +124,17 @@ public void deleteStudent(UUID studentId){
 
     @Override
     public StudentResponse updateStudent(UUID studentId, UpdateStudentRequest studentUpdated) {
-        validatePlanDetail(studentUpdated.paymentPlanName(), studentUpdated.paymentDay(), studentUpdated.extraClasses(), studentUpdated.classPrice());
+
         return studentRepository.findById(studentId)
                 .map(student -> {
+                    if (studentUpdated.getPaymentPlanName() == PaymentPlanName.BEGINNING_OF_MONTH) {
```

Comment:
Acá podríamos usar el método que usaste en el comentario anterior


```suggestion
                    if (planTypeIsBeginningOfMonth(studentUpdated.getPaymentPlanName())) {
```